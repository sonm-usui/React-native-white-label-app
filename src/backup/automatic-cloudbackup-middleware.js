// @flow

import {
  RECORD_HISTORY_EVENT,
  DELETE_HISTORY_EVENT,
} from '../connection-history/type-connection-history'
import { DELETE_CONNECTION_SUCCESS } from '../store/type-connection-store'
import { WALLET_ADDRESSES_REFRESHED } from '../wallet/type-wallet'
import {
  startAutomaticCloudBackup,
  setCloudBackupPending,
} from './backup-store'
import { BACKUP_STORE_STATUS } from './type-backup'
import { showUserBackupAlert } from '../connection-history/connection-history-store'
import { STORE_STATUS } from '../common/type-common'

const actionToRecord = [
  RECORD_HISTORY_EVENT,
  DELETE_HISTORY_EVENT,
  DELETE_CONNECTION_SUCCESS,
  WALLET_ADDRESSES_REFRESHED,
]

// TODO:KS Fix any type using `redux` provided Generic Types
const automaticCloudBackup = (store: any) => (next: any) => (action: any) => {
  const state = store.getState()
  // pass on the action first to other middleware in line
  const nextAction = next(action)

  // now dispatch action to backup wallet to the cloud
  if (actionToRecord.indexOf(action.type) > -1) {
    // NOTE: we got an action that requires backing up the wallet to the cloud
    // dispatch an action, that starts from beginning of middleware chain
    // we are dispatching a new action here
    if (state.backup.autoCloudBackupEnabled) {
      if (action.type === WALLET_ADDRESSES_REFRESHED) {
        if (
          state.wallet.walletAddresses.data.length === 0 &&
          state.wallet.walletAddresses.status === STORE_STATUS.IN_PROGRESS &&
          nextAction.walletAddresses.status === STORE_STATUS.SUCCESS
        ) {
          if (
            state.backup.cloudBackupStatus ===
              BACKUP_STORE_STATUS.CLOUD_BACKUP_LOADING ||
            state.backup.cloudBackupStatus ===
              BACKUP_STORE_STATUS.CLOUD_BACKUP_WAITING
          ) {
            store.dispatch(setCloudBackupPending(true))
          } else {
            store.dispatch(startAutomaticCloudBackup(action))
          }
        }
      } else {
        if (
          state.backup.cloudBackupStatus ===
            BACKUP_STORE_STATUS.CLOUD_BACKUP_LOADING ||
          state.backup.cloudBackupStatus ===
            BACKUP_STORE_STATUS.CLOUD_BACKUP_WAITING
        ) {
          store.dispatch(setCloudBackupPending(true))
        } else {
          store.dispatch(startAutomaticCloudBackup(action))
        }
      }
    } else {
      if (action.type === WALLET_ADDRESSES_REFRESHED) {
        if (
          state.wallet.walletAddresses.data.length === 0 &&
          state.wallet.walletAddresses.status === STORE_STATUS.IN_PROGRESS &&
          nextAction.walletAddresses.status === STORE_STATUS.SUCCESS
        ) {
          //NOTE: set state to indicate that a backup is really needed now
          store.dispatch(showUserBackupAlert(action))
        }
      }
    }
  }

  return nextAction
}

export default automaticCloudBackup
