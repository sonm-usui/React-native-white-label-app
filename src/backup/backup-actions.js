// @flow

import {
  CLOUD_BACKUP_SUCCESS,
  CLOUD_BACKUP_FAILURE,
  CLOUD_BACKUP_COMPLETE,
  VIEWED_WALLET_ERROR,
  SET_AUTO_CLOUD_BACKUP_ENABLED,
} from './type-backup'

export const cloudBackupSuccess = (lastSuccessfulCloudBackup: string) => ({
  type: CLOUD_BACKUP_SUCCESS,
  cloudBackupStatus: CLOUD_BACKUP_COMPLETE,
  lastSuccessfulCloudBackup,
})

export const cloudBackupFailure = (cloudBackupError: string) => ({
  type: CLOUD_BACKUP_FAILURE,
  cloudBackupStatus: CLOUD_BACKUP_FAILURE,
  cloudBackupError,
})

export const viewedWalletError = (viewedWalletError: boolean) => ({
  type: VIEWED_WALLET_ERROR,
  viewedWalletError,
})

export const setAutoCloudBackupEnabled = (autoCloudBackupEnabled: boolean) => ({
  type: SET_AUTO_CLOUD_BACKUP_ENABLED,
  autoCloudBackupEnabled,
})
