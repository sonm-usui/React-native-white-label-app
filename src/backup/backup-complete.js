// @flow

import React, { PureComponent } from 'react'
import { Image } from 'react-native'

import type { BackupCompleteProps, ReactNavigationBackup } from './type-backup'

import {
  Container,
  CustomView,
  CustomText,
  Icon,
  CustomHeader,
} from '../components'
import { backupCompleteRoute, settingsRoute } from '../common'
import { isBiggerThanVeryShortDevice } from '../common/styles'
import { color } from '../common/styles/constant'
import {
  BACKUP_COMPLETE_CLOSE_TEST_ID,
  BACKUP_COMPLETE_SUBMIT_BUTTON,
  BACKUP_COMPLETE_SUBMIT_BUTTON_TITLE,
} from './backup-constants'
import styles from './styles'
import { withStatusBar } from '../components/status-bar/status-bar'
import { appName } from '../external-imports'
import { Button } from '../components/buttons/button'

const transparentBands = require('../images/transparentBands.png')
const closeImage = require('../images/iconClose.png')
// TODO: use real asset
const successCheck = require('../images/successCheck.png')

export class BackupComplete extends PureComponent<BackupCompleteProps, void> {
  backupComplete = () => {
    const {
      navigation: { navigate },
    } = this.props

    return navigate(settingsRoute)
  }

  static navigationOptions = ({
    navigation: { navigate },
  }: ReactNavigationBackup) => ({
    header: () => (
      <CustomHeader
        backgroundColor={color.bg.fourteenth.color}
        largeHeader
        flatHeader
      >
        <CustomView style={[styles.headerSpacer]} />

        <CustomView style={[styles.headerSpacer]}>
          <Icon
            medium
            onPress={() => navigate(settingsRoute)}
            testID={BACKUP_COMPLETE_CLOSE_TEST_ID}
            iconStyle={[styles.headerIcon]}
            src={closeImage}
          />
        </CustomView>
      </CustomHeader>
    ),
    gestureEnabled: false,
    headerShown: true,
  })

  render() {
    return (
      <Container style={[styles.backupComplete]} safeArea>
        <Image source={transparentBands} style={[styles.backgroundImage]} />
        <Container style={[styles.wrapper]}>
          <Container>
            <CustomView center>
              <CustomText transparentBg center style={[styles.title]}>
                Backup complete
              </CustomText>
            </CustomView>
            <CustomView center>
              <Image source={successCheck} />
            </CustomView>
            <CustomView horizontalSpace>
              <CustomView
                verticalSpace
                doubleVerticalSpace={isBiggerThanVeryShortDevice ? true : false}
              >
                <CustomText
                  transparentBg
                  center
                  style={styles.backupCompleteText}
                >
                  If you ever have to start with a new installation of
                  {appName} you will need to recover from this saved backup
                  file.
                </CustomText>
              </CustomView>
              <CustomView
                doubleVerticalSpace={isBiggerThanVeryShortDevice ? true : false}
              >
                <CustomText
                  transparentBg
                  center
                  style={styles.backupCompleteText}
                >
                  You will be asked to enter your Recovery Phrase during setup
                  of your new {appName} application.
                </CustomText>
              </CustomView>
            </CustomView>
          </Container>
          <Button
            onPress={this.backupComplete}
            label={BACKUP_COMPLETE_SUBMIT_BUTTON_TITLE}
            buttonStyle={styles.submitButton}
            labelStyle={{
              color: color.bg.fourteenth.color,
              fontWeight: '600',
              fontSize: 18,
            }}
            testID={BACKUP_COMPLETE_SUBMIT_BUTTON}
          />
        </Container>
      </Container>
    )
  }
}

export const backupCompleteScreen = {
  routeName: backupCompleteRoute,
  screen: withStatusBar({ color: color.bg.fourteenth.color })(BackupComplete),
}
