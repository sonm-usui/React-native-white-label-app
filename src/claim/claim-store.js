// @flow
import { call, fork, put, select, spawn, takeEvery } from 'redux-saga/effects'
import moment from 'moment'
import type {
  Claim,
  ClaimAction, ClaimInfo,
  ClaimMap,
  ClaimReceivedAction,
  ClaimStorageFailAction,
  ClaimStorageSuccessAction,
  ClaimStore,
  MapClaimToSenderAction,
} from './type-claim'
import {
  CLAIM_RECEIVED,
  CLAIM_STORAGE_FAIL,
  CLAIM_STORAGE_SUCCESS,
  ERROR_CLAIM_HYDRATE_FAIL,
  HYDRATE_CLAIM_MAP,
  HYDRATE_CLAIM_MAP_FAIL,
  MAP_CLAIM_TO_SENDER,
} from './type-claim'
import type { CustomError, GenericObject } from '../common/type-common'
import { RESET } from '../common/type-common'
import {
  fetchPublicEntitiesForCredentials,
  getClaimHandleBySerializedClaimOffer,
  getCredentialInfo,
  updateClaimOfferState,
  updateClaimOfferStateWithMessage,
} from '../bridge/react-native-cxs/RNCxs'
import { CLAIM_STORAGE_ERROR } from '../services/error/error-code'
import {
  getAllConnection,
  getClaimForOffer,
  getClaimMap,
  getClaimOffer,
  getClaimOffers,
  getConnectionByUserDid,
  getConnectionHistory,
  getSerializedClaimOffer,
  getSerializedClaimOffers,
} from '../store/store-selector'
import { getHydrationItem, secureSet } from '../services/storage'
import { CLAIM_MAP } from '../common/secure-storage-constants'
import { updateMessageStatus } from '../store/config-store'
import { ensureVcxInitAndPoolConnectSuccess } from '../store/route-store'
import type { ClaimOfferPayload, SerializedClaimOffer } from '../claim-offer/type-claim-offer'
import { VCX_CLAIM_OFFER_STATE } from '../claim-offer/type-claim-offer'
import { saveSerializedClaimOffer } from '../claim-offer/claim-offer-store'
import type { Connection } from '../store/type-connection-store'
import { promptBackupBanner } from '../backup/backup-store'
import { captureError } from '../services/error/error-handler'
import { customLogger } from '../store/custom-logger'

export const claimReceived = (
  claim: Claim
): ClaimReceivedAction => ({
  type: CLAIM_RECEIVED,
  claim,
})

export const claimStorageSuccess = (
  messageId: string,
  claimId: string,
  issueDate: number
): ClaimStorageSuccessAction => ({
  type: CLAIM_STORAGE_SUCCESS,
  messageId,
  claimId,
  issueDate,
})

export const claimStorageFail = (
  messageId: string,
  error: CustomError
): ClaimStorageFailAction => ({
  type: CLAIM_STORAGE_FAIL,
  messageId,
  error,
})

export const mapClaimToSender = (
  claimUuid: string,
  senderDID: string,
  myPairwiseDID: string,
  logoUrl: string,
  issueDate: number,
  name: string,
  senderName?: string
): MapClaimToSenderAction => ({
  type: MAP_CLAIM_TO_SENDER,
  claimUuid,
  senderDID,
  myPairwiseDID,
  logoUrl,
  issueDate,
  name,
  senderName,
})

export const hydrateClaimMap = (claimMap: ClaimMap) => ({
  type: HYDRATE_CLAIM_MAP,
  claimMap,
})

export const hydrateClaimMapFail = (error: CustomError) => ({
  type: HYDRATE_CLAIM_MAP_FAIL,
  error,
})

export function* hydrateClaimMapSaga(): Generator<*, *, *> {
  try {
    const fetchedClaimMap = yield call(getHydrationItem, CLAIM_MAP)
    const connectionHistory = yield select(getConnectionHistory)
    const connections = yield select(getAllConnection)

    if (fetchedClaimMap) {
      const claimMap: ClaimMap = JSON.parse(fetchedClaimMap)
      // We added a new field issueDate, and already stored credentials
      // might not have this field populated, so when app is upgraded
      // and new code expects issueData, new code will find it undefined
      // so, here we are iterating on claimMap, and then adding issueDate
      // as current time
      for (const key in claimMap) {
        if (claimMap.hasOwnProperty(key) && !claimMap[key].issueDate) {
          claimMap[key].issueDate = moment().unix()
        }

        // We added a new field `name`, and already stored credentials
        // might not have this field populated, so when app is upgraded
        // new code will find it undefined
        // so, here we are iterating on connectionHistory, and then adding `name` to credential
        if (claimMap.hasOwnProperty(key) && !claimMap[key].name) {
          const event = connectionHistory.data.connections[
            claimMap[key].senderDID
            ].data.find(
            (event) =>
              event.originalPayload.type === CLAIM_STORAGE_SUCCESS &&
              claimMap[key].issueDate === event.originalPayload.issueDate
          )

          if (event) {
            claimMap[key].name = event.name
          }
        }
        if (claimMap.hasOwnProperty(key) && !claimMap[key].senderName) {
          const connection = connections[claimMap[key].myPairwiseDID]
          if (connection) {
            claimMap[key].senderName = connection.senderName
          }
        }
      }

      yield put(hydrateClaimMap(claimMap))
    }
  } catch (e) {
    captureError(e)
    customLogger.log(`hydrateClaimMapSaga: ${e}`)
    yield put(
      hydrateClaimMapFail({
        code: ERROR_CLAIM_HYDRATE_FAIL.code,
        message: `${ERROR_CLAIM_HYDRATE_FAIL.message}:${e.message}`,
      })
    )
  }
}

export function* claimReceivedSaga(
  action: ClaimReceivedAction
): Generator<*, *, *> {
  const { forDID, connectionHandle, uid, msg } = action.claim
  const vcxResult = yield* ensureVcxInitAndPoolConnectSuccess()
  if (vcxResult && vcxResult.fail) {
    return
  }

  const serializedClaimOffers: {
    string: SerializedClaimOffer[],
  } = yield select(getSerializedClaimOffers, forDID)

  if (connectionHandle != null) {
    const message = msg ? JSON.parse(msg) : {}

    // try to find correspondent Credential Offer using ~thread_id
    // it works for Aries protocol only
    if (message["~thread"]) {
      const offerId = message["~thread"]["thid"]
      const serializedClaimOffer = serializedClaimOffers[offerId]

      if (serializedClaimOffer){
        yield call(
          checkForClaim,
          serializedClaimOffer,
          connectionHandle,
          forDID,
          uid,
          offerId,
          msg
        )
        return
      }
    }

    // else we don't know for which claim offer this claim was sent
    // we receive a claim we only know claim message id,
    // and user id for which claim was sent
    // so, we take out all the claim offers for the connection
    // and then check each claim offer for update and download latest message
    // for each claim offer and see which claim offer received claim
    for (const offerId of Object.keys(serializedClaimOffers)) {
      // run each claim offer check in parallel and wait for all of them to finish
      const serializedClaimOffer = serializedClaimOffers[offerId]
      // $FlowFixMe
      yield call(
        checkForClaim,
        serializedClaimOffer,
        connectionHandle,
        forDID,
        uid,
        offerId,
        null
      )
    }
  }
}

export function* checkForClaim(
  serializedClaimOffer: SerializedClaimOffer,
  connectionHandle: number,
  userDID: string,
  uid: string,
  offerId: string,
  credentialMessage: ?string,
): Generator<*, *, *> {
  if (serializedClaimOffer.state === VCX_CLAIM_OFFER_STATE.ACCEPTED) {
    // if claim offer is already in accepted state, then we don't want to update state
    return
  }

  const { messageId } = serializedClaimOffer
  const claimOfferPayload = yield select(getClaimOffer, offerId)

  try {
    const claimHandle: number = yield call(
      getClaimHandleBySerializedClaimOffer,
      serializedClaimOffer.serialized
    )
    let vcxClaimOfferState: number

    if (credentialMessage) {
      vcxClaimOfferState = yield call(
        updateClaimOfferStateWithMessage,
        claimHandle,
        credentialMessage
      )
    } else {
      vcxClaimOfferState = yield call(
        updateClaimOfferState,
        claimHandle,
        credentialMessage
      )
    }


    if (vcxClaimOfferState === VCX_CLAIM_OFFER_STATE.ACCEPTED) {
      // once we know that this claim offer state was updated to accepted
      // that means that we downloaded the claim for this claim offer
      // and saved to wallet, now we need to know claim uuid and exact claim
      const vcxClaim: ClaimInfo = yield call(getCredentialInfo, claimHandle)
      const connection: ?Connection = yield select(
        getConnectionByUserDid,
        userDID
      )

      const issueDate = moment().unix()

      if (connection) {
        yield put(
          mapClaimToSender(
            vcxClaim.referent,
            connection.senderDID,
            userDID,
            connection.logoUrl,
            issueDate,
            claimOfferPayload.data.name,
            connection.senderName
          )
        )
      }

      yield put(claimStorageSuccess(serializedClaimOffer.messageId, vcxClaim.referent, issueDate))
      yield* updateMessageStatus([
        {
          pairwiseDID: userDID,
          uids: [messageId, uid],
        },
      ])
      yield put(promptBackupBanner(true))

      // since we asked vcx to update state, we should also update serialized state in redux
      // so that we don't go out of sync with vcx
      yield fork(
        saveSerializedClaimOffer,
        claimHandle,
        userDID,
        serializedClaimOffer.messageId
      )
    }
  } catch (e) {
    // we got error while saving claim in wallet, what to do now?
    captureError(e)
    yield put(
      claimStorageFail(serializedClaimOffer.messageId, CLAIM_STORAGE_ERROR(e))
    )
  }
}

export function* claimStoredSaga(): Generator<*, *, *> {
  // once we stored a new credential into the wallet we can update the cache containing
  // public entities (like Schemas, Credential Definitions) located on the Ledger.
  // This allows us to reduce the time taken for Proof generation (for the first credential usage) by
  // using already cached entities instead of queering the Ledger.
  // we even can not wait/handle the result of this function.
  // If querying failed we will query entities again during
  // proof generation and will get an error there if it fails again.
  yield spawn(fetchPublicEntitiesForCredentials)

  yield fork(saveClaimUuidMap)
}

export function* saveClaimUuidMap(): Generator<*, *, *> {
  try {
    const claimMap: ClaimMap = yield select(getClaimMap)
    yield call(secureSet, CLAIM_MAP, JSON.stringify(claimMap))
  } catch (e) {
    customLogger.log(`Failed to store claim uuid map:${e}`)
  }
}

export function* getClaim(claimOfferUuid: string): Generator<*, *, *> {
  const claimOffers = yield select(getClaimOffers)
  const claimOffer: ClaimOfferPayload = claimOffers[claimOfferUuid]
  if (!claimOffer){
    return
  }

  const { claimUuid, claim }: GenericObject = yield select(getClaimForOffer, claimOffer)
  if (!claim || !claimUuid){
    return
  }

  const identifier = claim.myPairwiseDID
  const vcxSerializedClaimOffer: SerializedClaimOffer = yield select(
    getSerializedClaimOffer,
    identifier,
    claimOfferUuid
  )

  const claimHandle = yield call(
    getClaimHandleBySerializedClaimOffer,
    vcxSerializedClaimOffer.serialized
  )

  return {
    claim,
    claimUuid,
    handle: claimHandle,
    vcxSerializedClaimOffer
  }
}

export function* watchClaimReceived(): any {
  yield takeEvery(CLAIM_RECEIVED, claimReceivedSaga)
}

export function* watchClaimStored(): any {
  yield takeEvery(CLAIM_STORAGE_SUCCESS, claimStoredSaga)
}

const initialState = {
  claimMap: {},
}

export default function claimReducer(
  state: ClaimStore = initialState,
  action: ClaimAction
) {
  switch (action.type) {
    case MAP_CLAIM_TO_SENDER:
      const {
        claimUuid,
        senderDID,
        myPairwiseDID,
        logoUrl,
        issueDate,
        name,
        senderName,
      } = action
      return {
        ...state,
        claimMap: {
          ...state.claimMap,
          [claimUuid]: {
            senderDID,
            myPairwiseDID,
            logoUrl,
            issueDate,
            name,
            senderName,
          },
        },
      }

    case HYDRATE_CLAIM_MAP:
      return {
        ...state,
        claimMap: action.claimMap,
      }

    case RESET:
      return initialState

    default:
      return state
  }
}
