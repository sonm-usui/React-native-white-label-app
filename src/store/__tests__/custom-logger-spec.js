// @flow
import {
  customLogger,
  PiiHiddenActionTransformer,
  PiiHiddenTransformer,
} from '../custom-logger'
import {
  getStore,
  walletBalance,
  walletAddresses,
  walletHistory,
} from '../../../__mocks__/static-data'
import { generateRecoveryPhraseSuccess } from '../../backup/backup-store'
import { mapClaimToSender } from '../../claim/claim-store'
import {
  walletBalanceRefreshed,
  tokenSentSuccess,
  walletAddressesRefreshed,
  walletHistoryRefreshed,
} from '../../wallet/wallet-store'

describe('Custom Logger', () => {
  it('Custom Logger should log to file', () => {
    customLogger.log('test log statement', ' asdfasdf ')
  })
})

describe('fn:PiiHiddenTransformer', () => {
  // Just having this single test for this transformer is enough
  // because we are testing whole redux store at once
  // and our mock data has all props for each and every module
  // and hence every path gets tested in single snapshot
  it('should hide all PII information with ****', () => {
    const mockStoreData = getStore().getState()
    expect(PiiHiddenTransformer(mockStoreData)).toMatchSnapshot()
  })
})

describe('fn:PiiHiddenActionTransformer', () => {
  it('should hide data Passphrase from action logging', () => {
    const action = generateRecoveryPhraseSuccess(
      getStore().getState().backup.passphrase
    )
    expect(PiiHiddenActionTransformer(action)).toMatchSnapshot()
  })

  it('should hide senderDID and pairwiseDID in MAP_CLAIM_TO_SENDER', () => {
    const action = mapClaimToSender(
      'claimUUID',
      'senderDID',
      'userDID',
      'logoUrl',
      123456789,
      'Test Claim',
      'Sender'
    )
    expect(PiiHiddenActionTransformer(action)).toMatchSnapshot()
  })

  it('should hide PII in WALLET_BALANCE_REFRESHED', () => {
    const action = walletBalanceRefreshed(walletBalance.data)
    expect(PiiHiddenActionTransformer(action)).toMatchSnapshot()
  })

  it('should hide PII in TOKEN_SENT_SUCCESS', () => {
    const action = tokenSentSuccess('123')
    expect(PiiHiddenActionTransformer(action)).toMatchSnapshot()
  })

  it('should hide PII in WALLET_ADDRESSES_REFRESHED', () => {
    const action = walletAddressesRefreshed(walletAddresses.data)
    expect(PiiHiddenActionTransformer(action)).toMatchSnapshot()
  })

  it('should hide PII in WALLET_HISTORY_REFRESHED', () => {
    const action = walletHistoryRefreshed(walletHistory.transactions)
    expect(PiiHiddenActionTransformer(action)).toMatchSnapshot()
  })
})
