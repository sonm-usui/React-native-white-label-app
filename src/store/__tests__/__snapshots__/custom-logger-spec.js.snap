// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fn:PiiHiddenActionTransformer should hide PII in TOKEN_SENT_SUCCESS 1`] = `
Object {
  "payment": Object {
    "error": null,
    "status": "SUCCESS",
    "tokenAmount": "****",
  },
  "type": "TOKEN_SENT_SUCCESS",
}
`;

exports[`fn:PiiHiddenActionTransformer should hide PII in WALLET_ADDRESSES_REFRESHED 1`] = `
Object {
  "type": "WALLET_ADDRESSES_REFRESHED",
  "walletAddresses": Object {
    "data": "****",
    "error": null,
    "status": "SUCCESS",
  },
}
`;

exports[`fn:PiiHiddenActionTransformer should hide PII in WALLET_BALANCE_REFRESHED 1`] = `
Object {
  "type": "WALLET_BALANCE_REFRESHED",
  "walletBalance": Object {
    "data": "****",
    "error": null,
    "status": "SUCCESS",
  },
}
`;

exports[`fn:PiiHiddenActionTransformer should hide PII in WALLET_HISTORY_REFRESHED 1`] = `
Object {
  "type": "WALLET_HISTORY_REFRESHED",
  "walletHistory": Object {
    "error": null,
    "status": "SUCCESS",
    "transactions": "****",
  },
}
`;

exports[`fn:PiiHiddenActionTransformer should hide data Passphrase from action logging 1`] = `
Object {
  "passphrase": "****",
  "status": "GENERATE_PHRASE_SUCCESS",
  "type": "GENERATE_RECOVERY_PHRASE_SUCCESS",
}
`;

exports[`fn:PiiHiddenActionTransformer should hide senderDID and pairwiseDID in MAP_CLAIM_TO_SENDER 1`] = `
Object {
  "claimUuid": "****",
  "issueDate": "****",
  "logoUrl": "****",
  "myPairwiseDID": "****",
  "name": "****",
  "senderDID": "****",
  "senderName": "****",
  "type": "MAP_CLAIM_TO_SENDER",
}
`;

exports[`fn:PiiHiddenTransformer should hide all PII information with **** 1`] = `
Object {
  "backup": Object {
    "backupWalletPath": "/Users/CoreSimulator/Devices/5F89AA14-B694-465F-82C3-838552AB4B85/data/Containers/Data/Application/4536A441-4B70-4A30-8DCC-5274E6781535/Documents/backup.zip",
    "cloudBackupPending": false,
    "error": null,
    "lastSuccessfulBackup": "2018-06-27T18:27:35+05:30",
    "lastSuccessfulCloudBackup": "2018-06-27T18:27:35+05:30",
    "passphrase": "****",
    "prepareBackupStatus": "PREPARE_BACKUP_SUCCESS",
    "showBanner": true,
    "status": "BACKUP_COMPLETE",
  },
  "claim": "****",
  "claimOffer": "****",
  "config": Object {
    "agencyDID": "****",
    "agencyUrl": "****",
    "agencyVerificationKey": "****",
    "isAlreadyInstalled": true,
    "isHydrated": false,
    "isInitialized": false,
    "paymentMethod": "sov",
    "poolConfig": "****",
    "showErrorAlerts": false,
    "vcxInitializationError": null,
    "vcxInitializationState": "VCX_INIT_SUCCESS",
    "vcxPoolInitializationError": null,
    "vcxPoolInitializationState": "VCX_INIT_POOL_SUCCESS",
  },
  "connections": Object {
    "connectionThemes": Object {
      "default": Object {
        "primary": "rgba(119, 119, 119, 1.0)",
        "secondary": "rgba(119, 119, 119, 0.6)",
      },
    },
    "data": undefined,
  },
  "deepLink": Object {
    "error": undefined,
    "isLoading": true,
    "tokens": "****",
  },
  "eula": Object {
    "isEulaAccept": true,
  },
  "history": Object {
    "data": Object {
      "connections": Object {
        "senderDID3": Object {
          "data": Array [
            Object {
              "action": "CONNECTED",
              "data": Array [
                Object {
                  "data": "2017-09-06T00:00:00+05:30",
                  "label": "Evernym",
                },
              ],
              "icon": Object {
                "default": "RelativeImageStub",
              },
              "id": "1",
              "name": "Enterprise name",
              "originalPayload": Object {},
              "remoteDid": "remoteDid",
              "status": "INVITATION_RECEIVED",
              "timestamp": "2017-09-06T00:00:00+05:30",
              "type": "INVITATION",
            },
          ],
          "newBadge": false,
        },
        "senderDID4": Object {
          "data": Array [
            Object {
              "action": "SHARED",
              "data": Array [
                Object {
                  "data": "2017-09-06T00:00:00+05:30",
                  "label": "Evernym",
                },
              ],
              "icon": Object {
                "default": "RelativeImageStub",
              },
              "id": "1",
              "name": "Enterprise name",
              "originalPayload": Object {},
              "remoteDid": "remoteDid",
              "status": "INVITATION_RECEIVED",
              "timestamp": "2017-09-06T00:00:00+05:30",
              "type": "INVITATION",
            },
          ],
          "newBadge": false,
        },
        "senderDID5": Object {
          "data": Array [
            Object {
              "action": "PENDING",
              "data": Array [
                Object {
                  "data": "2017-09-06T00:00:00+05:30",
                  "label": "Evernym",
                },
              ],
              "icon": Object {
                "default": "RelativeImageStub",
              },
              "id": "1",
              "name": "Enterprise name",
              "originalPayload": Object {},
              "remoteDid": "remoteDid",
              "status": "INVITATION_RECEIVED",
              "timestamp": "2017-09-06T00:00:00+05:30",
              "type": "INVITATION",
            },
          ],
          "newBadge": false,
        },
        "senderDID6": Object {
          "data": Array [
            Object {
              "action": "PROOF RECEIVED",
              "data": Array [
                Object {
                  "data": "2017-09-06T00:00:00+05:30",
                  "label": "Evernym",
                },
              ],
              "icon": Object {
                "default": "RelativeImageStub",
              },
              "id": "1",
              "name": "Enterprise name",
              "originalPayload": Object {
                "payload": Object {
                  "requester": Object {
                    "name": "bob",
                  },
                },
                "payloadInfo": Object {
                  "uid": "uidProofReceived",
                },
              },
              "remoteDid": "remoteDid",
              "status": "INVITATION_RECEIVED",
              "timestamp": "2017-09-06T00:00:00+05:30",
              "type": "INVITATION",
            },
          ],
          "newBadge": false,
        },
        "senderDID7": Object {
          "data": Array [
            Object {
              "action": "RECEIVED",
              "data": Array [
                Object {
                  "data": "2017-09-06T00:00:00+05:30",
                  "label": "Evernym",
                },
              ],
              "icon": Object {
                "default": "RelativeImageStub",
              },
              "id": "1",
              "name": "Enterprise name",
              "originalPayload": Object {
                "payload": Object {
                  "requester": Object {
                    "name": "bob",
                  },
                },
                "payloadInfo": Object {
                  "uid": "uidProofReceived",
                },
              },
              "remoteDid": "remoteDid",
              "status": "INVITATION_RECEIVED",
              "timestamp": "2017-09-06T00:00:00+05:30",
              "type": "INVITATION",
            },
          ],
          "newBadge": false,
        },
        "senderDID8": Object {
          "data": Array [
            Object {
              "action": "CLAIM OFFER RECEIVED",
              "data": Array [
                Object {
                  "data": "2017-09-06T00:00:00+05:30",
                  "label": "Evernym",
                },
              ],
              "icon": Object {
                "default": "RelativeImageStub",
              },
              "id": "1",
              "name": "Enterprise name",
              "originalPayload": Object {
                "payload": Object {
                  "requester": Object {
                    "name": "bob",
                  },
                },
                "payloadInfo": Object {
                  "uid": "uidProofReceived",
                },
              },
              "remoteDid": "remoteDid",
              "status": "INVITATION_RECEIVED",
              "timestamp": "2017-09-06T00:00:00+05:30",
              "type": "INVITATION",
            },
          ],
          "newBadge": false,
        },
      },
      "connectionsUpdated": true,
    },
    "isLoading": true,
  },
  "inAppNotification": Object {
    "notification": Object {
      "identifier": "receiverPairwiseDID1",
      "messageId": "messageId1",
      "messageType": "proofReq",
      "senderDID": "senderDID1",
      "senderImage": "https://localhost/image.png",
      "senderName": "Anonymous",
      "text": "Test wants you to share Address",
      "title": "You received a PROOF REQUEST",
    },
  },
  "invitation": "****",
  "ledger": Object {
    "fees": Object {
      "data": Object {
        "refreshTime": "2018-01-10T04:32:43+05:30",
        "transfer": "0.00001",
      },
      "error": null,
      "status": "SUCCESS",
    },
  },
  "lock": Object {
    "checkPinStatus": "CHECK_PIN_IDLE",
    "inRecovery": "false",
    "isAppLocked": false,
    "isLockEnabled": "true",
    "isTouchIdEnabled": true,
    "pendingRedirection": undefined,
    "showDevMode": false,
  },
  "offline": Object {
    "offline": false,
  },
  "openIdConnect": "****",
  "proof": "****",
  "proofRequest": "****",
  "pushNotification": Object {
    "error": undefined,
    "isAllowed": true,
    "isFetching": false,
    "isPristine": true,
    "navigateRoute": undefined,
    "notification": "****",
    "pendingFetchAdditionalDataKey": undefined,
    "pushToken": "****",
  },
  "question": Object {
    "data": "****",
    "storageStatus": "RESTORE_SUCCESS",
  },
  "restore": Object {
    "error": null,
    "passphrase": null,
    "restoreFile": Object {
      "fileName": "backup.zip",
      "fileSize": 22,
      "type": "application/zip",
      "uri": "file://data/Containers/Data/Application/021D78D1-044B-4874-BE8D-EC71781CA978/tmp/com.evernym.connectme.callcenter-Inbox/backup.zip",
    },
    "status": "none",
  },
  "route": Object {
    "currentScreen": "TabQRCodeScanner",
    "timeStamp": 1557756720914,
  },
  "sendlogs": Object {
    "encryptLogStatus": false,
  },
  "smsPendingInvitation": "****",
  "txnAuthorAgreement": Object {
    "haveAlreadySignedAgreement": false,
    "status": "IDLE",
    "taaAcceptedVersion": "",
    "text": "",
    "thereIsANewAgreement": false,
    "version": "",
  },
  "user": Object {
    "avatarName": "user-avatar.jpeg",
    "error": null,
    "isFetching": false,
    "userOneTimeInfo": "****",
  },
  "wallet": Object {
    "backup": Object {
      "backupPath": "test_backup_path",
      "encryptionKey": "****",
      "error": null,
      "fileStatus": "IDLE",
      "latest": "2017-09-06T00:00:00+05:30",
      "status": "IDLE",
    },
    "payment": Object {
      "error": null,
      "status": "SUCCESS",
      "tokenAmount": "****",
    },
    "walletAddresses": Object {
      "data": "****",
      "error": null,
      "status": "SUCCESS",
    },
    "walletBalance": Object {
      "data": "****",
      "error": null,
      "status": "SUCCESS",
    },
    "walletHistory": Object {
      "error": null,
      "status": "SUCCESS",
      "transactions": "****",
    },
  },
}
`;
